Bundler 2.3 docs and Bundler 2.3 blog post draft (#596)
rubygems/bundler-site: source/blog/2022-01-23-bundler-v2-3.html.markdown

---

title: "Bundler v2.3: Locking the version of Bundler itself"
date: 2022-01-23 16:07 UTC
tags:
author: David Rodríguez
author_url: https://github.com/deivid-rodriguez/
category: release
---

2021 saw a fair amount of development in the RubyGems & Bundler repositories. We
tried to release more often than ever to catch and fix bugs and distribute our

> improvements as early as possible users. That has led to 33 patch level versions

Reviewer: ```suggestion
improvements as early as possible to our users. That has led to 33 patch level versions
```

in the Bundler 2.x series released about a year ago.

> Our goal for Bundler 2.3 was to implement a long wanted feature of being able to

Reviewer: ```suggestion
Our goal for Bundler 2.3 was to implement a long-wanted feature of being able to
```

fully control the version of Bundler itself an application runs. There's a long
story with this feature, because it was shipped a few years ago in a manner that
was too strict and ended up causing more harm than good, so had to be partially
reverted.

## So, how did things work before Bundler 2.3?

Up until now, RubyGems would prefer to activate the version of Bundler recorded

> in the `Gemfile.lock` file if available, and would fallback to the highest

Reviewer: ```suggestion
in the `Gemfile.lock` file if available, and would fall back to the highest
```
